### ISSUE TITLE
"jump in" node's text as (part of) view tab label

### ISSUE THREAD

**Issue by EdoFro** [freeplane/freeplane#1484]
> One of the uses of the "New map view" option is that we can focus large maps on several branches in separate tabs.
> Renaming the tab is a good option to identify which branch the tabs represent.
> But the better and easier way is that when we apply the "jump in" option on a node, the textual content of the jumped node will be shown automatically on the tab and we don't need to change the name of the tab manually.
> 
> If this feature is possible, please add it.

_Originally posted by @ajaberi1360 in https://github.com/freeplane/freeplane/issues/1476#issuecomment-1769156530_

**Comment by EdoFro** [comment]
also mentioned here:
https://github.com/freeplane/freeplane/pull/1353#issuecomment-1773688421

**Comment by EdoFro** [comment]
@dpolivaev,
Can you please add a "feature request" label?
Thanks

**Comment by euu2021** [comment]
Yes, I think it would be specially useful for people who use one single map with multiple views. For people who use several different maps, it may get confusing, because the name of the tab is useful to identify the map.

I use one single map, so, for me, if the tab gets the name of the current root (JumpIn root), it's more useful than having, by default, the name of the original Root with a number (current behavior). I mean, if I JumpIn into a node I want it to be the new central subject, so it makes sense if it's the name of the tab. 

On the other hand, if I set the name of the tab manually, I don't want it to change automatically with JumpIns.

Perhaps, the least compromising short term solution is to create an API method that sets the name of the tab, and another that informs if the tab was renamed by right clicking on it. So, the user can create a script, to be used as the Jump In hotkey, that does that:

- if (the current tab wasn't named by right clicking) {set the text of the selected node as the name of the current tab}
- else {do nothing about the tab name}
- Jump In

This would also give more flexibility for the user to define the behavior any way he needs. For example, if he wants to set the details (instead of the node core) as the tab name.

**Comment by aaa1386** [comment]
Yes, it is true that the name of the tab is useful for identifying the map, but it is also useful for the node on which it is jumped, provided that the pin used in the mental map for the jumped node is also used in the tab.  Another sign can also be used.

In my opinion, it  meet all the demands of different users  
 1- Those who use one single mental map 2- Those who use several mental map files instead of one map.

It seems that the "New map view" option is only useful when we use the "Jump In" option in the mental map in the new view. It makes sense if the name of the jumped node is entered in the label of the automatic tab, especially if there is a sign that shows that the existing tab is a jumped node.
What is the point of opening the map in a new view without using the "jump in" option? I do not understand.
It will be more complete if the following ideas are also used.
Create an option to save the opened and defined tabs, so that the next time we open the mind map file, the previously opened tabs will be opened in the same way.

**Comment by euu2021** [comment]
> It will be more complete if the following ideas are also used.
Create an option to save the opened and defined tabs, so that the next time we open the mind map file, the previously opened tabs will be opened in the same way.

This would be very useful to me. I usually have more than 10 map views opened, in specific arrangements, that are usually the same, every day. So, when I need to close FP, I need to reopen all tabs and rearrange then, which is annoying. Now, it will be even more annoying, because I will need to rename every tab 🤣🤣

**Comment by EdoFro** [comment]
> > It will be more complete if the following ideas are also used.
> > Create an option to save the opened and defined tabs, so that the next time we open the mind map file, the previously opened tabs will be opened in the same way.
> 
> This would be very useful to me. I usually have more than 10 map views opened, in specific arrangements, that are usually the same, every day. So, when I need to close FP, I need to reopen all tabs and rearrange then, which is annoying. Now, it will be even more annoying, because I will need to rename every tab 🤣🤣

I use the Hybernate feature in windows for power my notebook off. This way I can easily continue my work the next day.

**Comment by EdoFro** [comment]
I helped implementing the new tab renaming feature, but I don’t think I’m going to add more functionalities to it. It is very time consuming and I have other priorities.
If someone else wants to give it a try, I can explain what I learned in the process.

Regards,

eso

**Comment by euu2021** [comment]
> I use the Hybernate feature in windows for power my notebook off. This way I can easily continue my work the next day.

Thanks for the tip. Yes, I do that, too. But, sometimes, I need to restart FP, in order to use the latest version, or restart the PC because of the Windows Updates and other software updates.

> I helped implementing the new tab renaming feature, but I don’t think I’m going to add more functionalities to it. It is very time consuming and I have other priorities.

It is very useful. Thanks!

**Comment by aaa1386** [comment]
> I helped implementing the new tab renaming feature, ...

Thanks for the effort you put into writing the tab renaming idea.
If possible, please include the idea raised in this topic in the software.("jump in" node's text as (part of) view tab label #1484)
Of course, I understand the difficulty and time-consuming nature of making changes to Freeplane software.

**Comment by aaa1386** [comment]
see : https://github.com/freeplane/freeplane/discussions/2882

**Comment by dpolivaev** [comment]
https://github.com/freeplane/freeplane/releases/tag/release-1.13.3-pre06
Options to show root node text or bookmark titles on tab titles and map menu labels
  - Preferences…->Appearance->Look and Feel->Tab titles
  - Preferences…->Appearance->Look and Feel->Maximum tab title length
